Funcky/Monads/Either/Either.Convenience.cs

```diff
@@ -16,6 +16,13 @@ namespace Funcky.Monads;
         return this;
     }
 
+    /// <summary>Performs a side effect when the either is left and returns the either value again.</summary>
+    public Either<TLeft, TRight> InspectLeft(Action<TLeft> inspector)
```

Comment:
Discussion: We could also go for an alternative design where we use multiple overloads of the `Inspect` method and use the argument name to distinguish the overloads:

```cs
Inspect(left: x => ...);
Inspect(right: x => ...);
Inspect(left: ..., right: ...);
```

This has the advantage that we can have an inspect that inspects both states at the same time.
But I think distinguishing overloads by argument names is not very user friendly.